### PR TITLE
TrafficGate Bid Adapter: add new bid adapter

### PR DESCRIPTION
**Pull request by vlaktion**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by vlaktion** [comment]
https://github.com/prebid/prebid.github.io/pull/3711

**Comment by vlaktion** [comment]
https://github.com/prebid/prebid-server/pull/2226

File: modules/trafficgateBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import { getWindowLocation, deepAccess } from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'trafficgate';
+const URL = '//[HOST].bc-plugin.com/?c=o&m=multi';
```
Comment (Fawke) [inline modules/trafficgateBidAdapter.js:6]: Can you please hardcode this to https?

**Review by Fawke — COMMENTED** [review]
Hi @vlaktion,

I tried to test the example that you gave in your md file, but I didn't get any bid response back. Can you please provide a working example?

File: modules/trafficgateBidAdapter.js (hunk 1 above)
Comment (vlaktion) [reply modules/trafficgateBidAdapter.js:6]: Hello. What if site use http only? It will error 'https request over http'?

**Comment by vlaktion** [comment]
Hello. Yes, we have test environment. You need send parameter "placementId": 0, it will give test response. vudha.bc-plugin.com it is one of our partners. Example:
URL:
https://vudha.bc-plugin.com/?c=o&m=multi
Body:
{
    "bidId": "9ec5b177515ee2e5",
    "bidder": "trafficgate",
    "language": "en-EN",
    "secure": 1,
    "host": "example.com",
    "page": "/page",
    "placements": [{
        "placementId": 0,
        "bidId": "9ec5b177515ee2e5",
        "traffic": "video"
    }]
  }

File: modules/trafficgateBidAdapter.js (hunk 1 above)
Comment (Fawke) [reply modules/trafficgateBidAdapter.js:6]: In my experience, sending request to secure endpoints from http webpages doesn't cause any issue. You can test it out locally if you want, but I ran the [hello world page](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html) with a local web server running http, and it did made the request to the secure appnexus endpoint.

Also, I see that the endpoint doesn't have a fixed host. Is there any specific reason to have a "variable" host name?

**Comment by Fawke** [comment]
> Hello. Yes, we have test environment. You need send parameter "placementId": 0, it will give test response. vudha.bc-plugin.com it is one of our partners. Example: URL: https://vudha.bc-plugin.com/?c=o&m=multi Body: { "bidId": "9ec5b177515ee2e5", "bidder": "trafficgate", "language": "en-EN", "secure": 1, "host": "example.com", "page": "/page", "placements": [{ "placementId": 0, "bidId": "9ec5b177515ee2e5", "traffic": "video" }] }

Sorry, I tried to click the url: https://vudha.bc-plugin.com/?c=o&m=multi, but it's not working. Is there any way for me to test this adapter locally?

**Comment by vlaktion** [comment]
@Fawke Hi. Did you send a POST query?
Screenshot 2022-06-03 at 11 08 37

**Comment by Fawke** [comment]
Hi @vlaktion, 

Can you please hardcode your endpoint url to `https`? Everything else looks good.

**Comment by vlaktion** [comment]
Done

**Review by Fawke — APPROVED** [review]
LGTM

**Pull request merged**